File: EEntityCore.DB/EEntityCore.DB.MSSQL/Interfaces/IDBViewDefinitionPlugIn.vb


Namespace Interfaces


    Public Interface IDBViewDefinitionPlugIn


        ''' <summary>
        ''' Return the names of tables this view depend on
        ''' </summary>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Function GetReferencedTableNames() As List(Of String)

        Function GetViewName() As String

        Function GetFillSQL() As String

    End Interface



End Namespace

File: EEntityCore.DB/EEntityCore.DB.MSSQL/Interfaces/IDBViewDefinitionPlugIn.cs
using System.Collections.Generic;

namespace EEntityCore.DB.MSSQL.Interfaces
{
    public interface IDBViewDefinitionPlugIn
    {


        /// <summary>
        /// Return the names of tables this view depend on
        /// </summary>
        /// <returns></returns>
        /// <remarks></remarks>
        List<string> GetReferencedTableNames();
        string GetViewName();
        string GetFillSQL();
    }
}